File: 3-leetcode/01-easy/js/205-isomorphic-strings.ts
const isIsomorphic = (s: string, t: string): boolean => {
  const length = s.length;
  const sArr = [];
  const tArr = [];

  for (let i = 0; i < length; i++) {
    const sCode = s.charCodeAt(i);
    const tCode = t.charCodeAt(i);

    if (sArr[sCode] !== tArr[tCode]) {
      return false;
    }

    sArr[sCode] = i;
    tArr[tCode] = i;
  }

  return true;
};

File: 3-leetcode/01-easy/cpp/205-isomorphic-strings.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    bool isIsomorphic(string s, string t)
    {
        vector<int> sArr(128, -1);
        vector<int> tArr(128, -1);

        int length = s.length();

        for (int i = 0; i < length; i++)
        {
            if (sArr[s[i]] != tArr[t[i]])
            {
                return false;
            }

            sArr[s[i]] = i;
            tArr[t[i]] = i;
        }

        return true;
    }
};
